internal/storage/references/registry_reference.go

```diff
@@ -40,6 +54,8 @@ func RegistryImageReferenceFromRaw(rawNamed reference.Named) RegistryImageRefere
 //
 // It is only intended for communication with OUT-OF-PROCESS APIs,
 // like registry references provided by CRI by Kubelet.
+// It will modify the reference if both digest and tag are specified, stripping the tag and leaving the digest.
+// It will also verifies the image is not only a name. If it is only a name, the function errors.
```

Comment:
The function does not fail in such a situation: The `TagNameOnly` call turns name-only references into _name_`:latest`